fuel-tx/src/lib.rs

```diff
@@ -100,3 +100,21 @@ pub use transaction::consensus_parameters::default_parameters;
 
 #[cfg(feature = "alloc")]
 pub use contract::Contract;
+
+/// Trait extends the functionality of the `ContractId` type.
+pub trait ContractIdExt {
+    /// Creates an `AssetId` from the `ContractId` and `sub_id`.
+    fn asset_id(&self, sub_id: &Bytes32) -> AssetId;
```

Comment:
Is it worth reusing `AssetID` here, or maybe creating a type/type alias for `SubId`?